command/check.go

```diff
@@ -59,10 +61,45 @@ func (c *CheckCommand) Run(ctx context.Context, originalArgs []string) error {
 		return fmt.Errorf("expected exactly one argument, got %d", got)
 	}
 
-	inFile := args[0]
-	m, err := parseYAMLFile(inFile)
+	input := args[0]
+
+	fileInfo, err := os.Stat(input)
+	if err != nil {
+		return fmt.Errorf("input not found %w", err)
+	}
+
+	skipRoot := false
+	if fileInfo.IsDir() {
```

Comment:
I think there's a few things here:

1. We probably want to restrict to `.yml` and `.yaml` files only. I imagine some folders will contain more than just YAML files.

1. If we're going to add directory support, we need to plumb in through lower in the stack and make it available in all the commands (not just check).

1. The current implementation doesn't summarize the total findings for the exit code; it only returns the exit code for the _last_ file checked.

1. This will need test coverage; I'm especially weary about how this will work on Windows systems.